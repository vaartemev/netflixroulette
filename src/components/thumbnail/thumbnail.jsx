import React from 'react';
import PropTypes from 'prop-types';

import './thumbnail.scss';

export const Thumbnail = ({ src, className, handleOnClick }) => {
  return <img src={src} alt="" className={className} onClick={handleOnClick} />;
};

Thumbnail.propTypes = {
  src: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};
