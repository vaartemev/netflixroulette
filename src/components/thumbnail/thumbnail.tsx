import React from 'react';

import './thumbnail.scss';

interface ThumbnailProps {
  src: string;
  className: string;
}

export const Thumbnail: React.FC<ThumbnailProps> = ({ src, className }) => {
  return <img src={src} alt="" className={className} />;
};
