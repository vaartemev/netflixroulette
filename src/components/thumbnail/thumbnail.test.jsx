import { React } from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Thumbnail } from './thumbnail';

describe('Thumbnail', () => {
  const props = {
    src: 'string2',
    className: 'string',
  };

  const output = shallow(<Thumbnail {...props} />);

  it('renders correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
