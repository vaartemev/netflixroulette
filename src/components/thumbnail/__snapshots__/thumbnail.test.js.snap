// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail renders correctly 1`] = `
<img
  className="string"
  src="string2"
/>
`;
